source/elasticity.cc

```diff
@@ -129,9 +129,12 @@ ElasticityProblem<dim, spacedim>::make_grid()
     {
       GridIn<spacedim> gi;
       gi.attach_triangulation(tria);
-      // std::ifstream infile(par.name_of_grid);
-      const std::string infile(par.name_of_grid);
+#ifdef DEAL_II_WITH_GMSH_API
+      std::string infile(par.name_of_grid);
       Assert(!infile.empty(), ExcIO());
```

Comment:
This assert does not make sense for a string. The check on the existence of the file is done inside the function below for the gmsh api. The assert should be moved to the other branch of the `#ifdef`